drivers/ethernet/eth_stm32_hal.c

```diff
@@ -132,16 +135,16 @@ static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef *heth,
 #endif /* CONFIG_SOC_SERIES_STM32H7X */
 }
 
-static inline void disable_mcast_filter(ETH_HandleTypeDef *heth)
+static inline void setup_mac_filter(ETH_HandleTypeDef *heth)
 {
 	__ASSERT_NO_MSG(heth != NULL);
 
 #if defined(CONFIG_SOC_SERIES_STM32H7X)
 	ETH_MACFilterConfigTypeDef MACFilterConf;
 
 	HAL_ETH_GetMACFilterConfig(heth, &MACFilterConf);
-	MACFilterConf.HashMulticast = DISABLE;
-	MACFilterConf.PassAllMulticast = ENABLE;
+	MACFilterConf.HashMulticast = ENABLE;
+	MACFilterConf.PassAllMulticast = DISABLE;
```

Comment:
```suggestion
#ifdef CONFIG_ETH_MULTICAST_FILTER
  	MACFilterConf.HashMulticast = ENABLE;
  	MACFilterConf.PassAllMulticast = DISABLE;
#else 
        MACFilterConf.HashMulticast = DISABLE;
  	MACFilterConf.PassAllMulticast = ENABLE;
#endif  /* CONFIG_ETH_MULTICAST_FILTER */
```

Please add CONFIG_ETH_MULTICAST_FILTER in KConfig also. It would be better to make it controlled by KConfig.